Frameworks/include/NSEnumeratorInternal.h

```diff
@@ -18,6 +18,11 @@
 
 #include <Foundation/NSEnumerator.h>
 
+#define NS_COLLECTION_THROW_ILLEGAL_KVO(KEYPATH)                                                                                         \
+    do {                                                                                                                                 \
+        [NSException raise:NSInvalidArgumentException format:@"%@ is not supported. Key path: %@", NSStringFromSelector(_cmd), KEYPATH]; \
```

Comment:
nit: no shouty `KEYPATH`.
Please add `-[%hs %hs]` with `object_getClassName(self), sel_getName(_cmd)`; this will add the class (obvs.) and avoid the NSString intermediary for `_cmd`.